hieradata/class/share_sale_publisher.yml

```diff
@@ -0,0 +1,4 @@
+---
+
+govuk::node::s_base::apps:
+  - share_sale_publisher
```

Comment:
Are we really adding new machines just for this app (that question isn't necessarily directed at you @mathildathompson)? If so, should the machines be named `share-sale-publisher-n.backend`? This isn't something we've done before. There will also need to be a node definition added (`modules/govuk/manifests/node/s_foo.pp`) and host entries for these machines (hieradata).